### ISSUE TITLE
Upgrade to `reek` 5.6.0

### ISSUE THREAD

**Issue by lloydwatkin** [whitesmith/rubycritic#341]
I wasn't able to use `rubycritic` after upgrading my `mysql2` gem (to handle BigDecimal deprecation warnings on ruby 2.6.5). This was due to `reek` requiring `parser<2.7`. `reek` has today updated their parser requirements to fix this issue, see: https://github.com/troessner/reek/blob/master/CHANGELOG.md#560-2020-01-20

**Comment by lloydwatkin** [comment]
```
Bundler could not find compatible versions for gem "parser":
  In snapshot (Gemfile.lock):
    parser (= 2.7.0.1)

  In Gemfile:
    i18n-tasks (~> 0.9.23) was resolved to 0.9.29, which depends on
      parser (>= 2.2.3.0)

    rubycritic was resolved to 4.3.1, which depends on
      reek (~> 5.0, < 6.0) was resolved to 5.5.0, which depends on
        parser (>= 2.5.0.0, != 2.5.1.1, < 2.7)

    rubocop was resolved to 0.78.0, which depends on
      parser (>= 2.6)

    rubycritic was resolved to 4.3.1, which depends on
      parser (~> 2.6.0)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.

**Comment by nunosilva800** [comment]
I think we can just relax the version constraint on `parser`.

Can you do a PR to change:
`spec.add_runtime_dependency 'parser', '~> 2.6.0'` 
to 
`spec.add_runtime_dependency 'parser', '>= 2.6.0'`
here: https://github.com/whitesmith/rubycritic/blob/master/rubycritic.gemspec#L36

Thanks 😄
